expression/builtin_cast_vec.go

```diff
@@ -583,11 +585,30 @@ func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinCastDurationAsDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var err error
+	if err = b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	res := result.GoDurations()
+	dec := b.tp.Decimal
+	for i, v := range res {
+		if result.IsNull(i) {
+			continue
+		}
+		dec, err := types.CheckFsp(dec)
+		if err != nil {
+			return err
+		}
+		n := time.Date(0, 0, 0, 0, 0, 0, 0, time.Local)
```

Comment:
Can we define a zero `Time` outside this loop for example `var zeroT time.Time` and reset it in each loop and use it to do the add operation so that we don't need to create a new `Time` in each loop?

Reply:
ok , l have modified it.

Reply:
And please update the benchmark result after modify.

Reply:
Got it. I have updated the benchmark result.